superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -124,6 +205,145 @@ export default class ResultSet extends React.PureComponent<
     }
   }
 
+  handleOnChangeAutoComplete = () => {
+    this.setState({ datasetToOverwrite: {} });
+  };
+
+  handleOverwriteDataset = async () => {
+    const { sql, results, dbId } = this.props.query;
+    const { datasetToOverwrite } = this.state;
+
+    updateDatset(
+      datasetToOverwrite.datasetId,
+      dbId,
+      sql,
+      results.selected_columns.map(d => ({ column_name: d.name })),
+      true,
+    )
+      .then(() => {
+        exploreChart({
+          ...EXPLORE_CHART_DEFAULT,
+          datasource: `${datasetToOverwrite.datasetId}__table`,
+          all_columns: results.selected_columns.map(d => d.name),
+        });
+      })
+      .catch(() => {
+        this.props.actions.addDangerToast(
+          t('An error occurred overwriting dataset'),
+        );
+      });
+
+    this.setState({
+      showSaveDatasetModal: false,
+      shouldOverwriteDataSet: false,
+      datasetToOverwrite: {},
+      newSaveDatasetName: `${this.props.query.tab} ${moment().format(
+        'MM/DD/YYYY HH:mm:ss',
+      )}`,
+    });
+  };
+
+  handleSaveInDataset = () => {
+    // if user wants to overwrite a dataset we need to prompt them
+    if (
+      this.state.saveDatasetRadioBtnState ===
+      DatasetRadioState.OVERWRITE_DATASET
+    ) {
+      this.setState({ shouldOverwriteDataSet: true });
+      return;
+    }
+
+    const { schema, sql, dbId, templateParams } = this.props.query;
+    const selectedColumns = this.props.query?.results?.selected_columns || [];
+
+    this.props.actions
+      .createDatasource({
+        schema,
+        sql,
+        dbId,
+        templateParams,
+        datasourceName: this.state.newSaveDatasetName,
+        columns: selectedColumns,
+      })
+      .then((data: { table_id: number }) => {
+        exploreChart({
+          datasource: `${data.table_id}__table`,
+          metrics: [],
+          groupby: [],
+          time_range: 'No filter',
+          viz_type: 'table',
+          all_columns: selectedColumns.map(c => c.name),
+          row_limit: 1000,
+        });
+      })
+      .catch(() => {
+        this.props.actions.addDangerToast(
+          t('An error occurred saving dataset'),
+        );
+      });
+
+    this.setState({
+      showSaveDatasetModal: false,
+      newSaveDatasetName: `${this.props.query.tab} ${moment().format(
+        'MM/DD/YYYY HH:mm:ss',
+      )}`,
+    });
```

Comment:
There's seem to be some repetitive code between Override and Save as New. Is there a chance for abstraction?